### PR TITLE
[stable/etcd-operator] Add cluster.restore, cluster.pod, nodeSelector & update

### PR DESCRIPTION
**Pull request by mikesplain**
Added cluster.restore, cluster.pod, and nodeSelctors to both the etcd-operator pod and etcd cluster pods.

I've also updated the default version of the operator and etcd.

### CODE REVIEW

**Comment by lachie83** [comment]
cc @chancez Thoughts on this?

**Comment by lachie83** [comment]
@mikesplain Can you please share the reference of this being supported upstream in the etcd cluster definition?

**Comment by chancez** [comment]
I'm not sure about this. Why not just add the options that the operator supports via the TPR directly into the template?

**Comment by lachie83** [comment]
Yeah that's what I would suggest too. Although I don't see any reference in the upstream example that we aren't yet covering

**Comment by mikesplain** [comment]
@chancez @lachie83 There are a few examples [here](https://github.com/coreos/etcd-operator/blob/a567f269c6aaa32614e24d2ed34aafadcb8794bd/doc/user/spec_examples.md#three-members-cluster-with-node-selector-and-anti-affinity), as above.  Currently `spec.pod.*` can't be set in this chart. 

I don't mind switching it to add `pod.*` instead but I've seen quite a few of these types of options not present in charts (simply because software moves fast) so I figured by adding more of a catch all, it'll prevent having to open a PR every time a chart's application adds a new feature or flag.

**Comment by chancez** [comment]
@lachie83 it should basically cover all things in this struct: https://github.com/coreos/etcd-operator/blob/master/pkg/spec/cluster.go#L66 which in the example given, pod affinity is one thing to support, among others.

**Comment by lachie83** [comment]
Thanks @mikesplain. Let me review and get back to you.

File: stable/etcd-operator/values.yaml [hunk 1]
```diff
@@ -31,3 +31,4 @@ cluster:
       storageType: PersistentVolume
       pv:
         volumeSizeInMB: 512
+  extraArgs: {}
```
Comment (lachie83) [inline stable/etcd-operator/values.yaml]: Can you put reference to the upstream link with some examples and also add some commented out examples here?

**Comment by lachie83** [comment]
Perfect thanks @chancez

**Comment by chancez** [comment]
And I'm still of the opinion that I'd rather see the helm chart explicitly enumerate available options rather than using the generic extraArgs, personally.

**Comment by lachie83** [comment]
I tend to agree with @chancez . Can you scope that out in this PR?

**Comment by mikesplain** [comment]
@chancez @lachie83 I agree with your points, however would you mind having both?  I don't mind adding the other args but I hoping to add an option to allow users to work around explicit options not being present.

**Comment by chancez** [comment]
@mikesplain I personally think that that's best handled by forking the chart. I dislike trying to make charts overly extensible, because it means it can be difficult to handle upgrades. 

If a user specifies an option in extraArgs because it's added to the operator, then we add that new option to the template, then it will be specified twice. This is bad for TPRs in particular because the schema isn't validated on submission, so the upgrade will succeed, but the TPR will be broken most likely.

**Comment by lachie83** [comment]
I'm going to stand behind @chancez on this one. 

Best way forward is to either fork and implement, or implement specific key vals. 

How would you like to proceed @mikesplain

**Comment by mikesplain** [comment]
Alrighty, I'll modify this to just be those values then.

**Comment by lachie83** [comment]
Thanks @mikesplain . Appreciate your understanding

**Comment by mikesplain** [comment]
Absolutely, you guys know this stuff much better than I, I'm simply trying to help others work around road blocks I've come up against frequently as of late.

I am curious though, what's the intended path for forking @lachie83 @chancez?

Background: We use this chart as a dependency for one of our applications.  When a developer finds a chart they want to use is missing a feature they need, they fork it, modify it and PR it back upstream.  That's super easy, but the problem comes when you want to use the fork temporarily until the PR is merged, etc.  Since dependency charts have to either be in the charts directory or in a chart repository as a package (based on my understanding), it requires forkers to setup a bit of other infrastructure to use (s3 or gsutil).  I've setup https://github.com/technosophos/helm-plugins for us against our fork of this repo but it's kind of a hack... I was wondering if you have a suggested or best practice for forkers to utilize their forks without having to duplicate the [merge and release process](https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#merge-approval-and-release-process) here? I appreciate your advice while I update the PR.

**Comment by chancez** [comment]
If I have to fork, I just change my dependency to reference the chart on my local file system using the `file://` syntax. If you don't have control over the chart which has a dependency, then you fork that one too, and just modify it's dependency. It's not the best but it's not that bad either.

**Comment by mikesplain** [comment]
@chancez @lachie83 I've updated the PR as suggested and added a few other things. Let me know if you have additional feedback.

**Comment by unguiculus** [comment]
Requires a chart version bump.

@viglesiasce @lachie83 Is this good to go?

**Comment by lachie83** [comment]
LGTM. Please bump the chart version

**Review by lachie83 — APPROVED** [review]

**Comment by mikesplain** [comment]
Sorry for the delay, version bumped @lachie83 @unguiculus

**Comment by lachie83** [comment]
@mikesplain we need to bump again. The rebase nullified your change.

**Comment by mikesplain** [comment]
@lachie83 Ugh, sorry about that. Had a bad git day yesterday.... Fixed now.

**Comment by lachie83** [comment]
LGTM

**Pull request merged**